### ISSUE TITLE
bug of multilable loss in v0.4

### ISSUE THREAD

**Issue by lee-junjie** [pytorch/pytorch#9301]
```python
>>> import torch
>>> import torch.nn.functional as F
>>> import numpy as np
>>> outputs = torch.ones(2, 5) * np.log(1/99)
>>> preds = F.sigmoid(outputs)
>>> labels = torch.zeros(2, 5)
>>> labels[:,0] = 1
>>> preds
tensor(1.00000e-03 *
       [[10.0000, 10.0000, 10.0000, 10.0000, 10.0000],
        [10.0000, 10.0000, 10.0000, 10.0000, 10.0000]])
>>> labels
tensor([[ 1.,  0.,  0.,  0.,  0.],
        [ 1.,  0.,  0.,  0.,  0.]])
>>> -torch.sum(labels * torch.log(preds) + (1-labels) * torch.log(1-preds)) / outputs.size(0)   # the expected result of multilabel loss
tensor(4.6454)
>>> -torch.mean(labels * torch.log(preds) + (1-labels) * torch.log(1-preds))       # a wrong way, should use sum(not mean) for binary losses of different classes
tensor(0.9291)
>>> F.multilabel_soft_margin_loss(outputs, labels) # same as above
tensor(0.9291)
>>> F.multilabel_soft_margin_loss(outputs, labels, size_average=False)     # expected behavior, but wrong for loss, should average loss from samples in a batch
tensor(9.2907)
>>> F.multilabel_soft_margin_loss(outputs, labels, size_average=False) / outputs.size(0)    # correct
tensor(4.6454)
```

doc: [https://pytorch.org/docs/stable/nn.html?highlight=multilabel_soft_margin_loss#torch.nn.MultiLabelSoftMarginLoss](https://pytorch.org/docs/stable/nn.html?highlight=multilabel_soft_margin_loss#torch.nn.MultiLabelSoftMarginLoss)

I think the default behavior of multilabel_soft_margin_loss is incorrect: current 0.4 version will **`average`** binary losses of classes in a sample. From the doc, we should use `sum`, not `average`.

**Comment by SsnL** [comment]
cc @li-roy

**Comment by li-roy** [comment]
It is not. @weiyangfb since you already have #9965 open, can you fix this as well? From a cursory glance, I'm guessing this bug has been there since size_average was introduced, so tests might be wrong as well.

**Comment by weiyangfb** [comment]
@li-roy Yeah, I will fix this.

**Comment by weiyangfb** [comment]
closing because it should be fixed in https://github.com/pytorch/pytorch/pull/9965
